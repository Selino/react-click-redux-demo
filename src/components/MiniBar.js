import React from "react"
import { Flip } from "react-reveal"

export default function MiniBar(props) {
  const myBarBackGround = {
    backgroundColor: props.backGroundColor,
    width: "60px",
    height: "16px",
    borderRadius: "2px",
    overflow: "hidden",
    display: "inline-block",
    marginLeft: ".5rem",
    verticalAlign: "text-bottom",
  }
  const myBarForGround = {
    backgroundColor: props.forGroundColor,
    width: props.Perc + "%",
    height: "100%",
  }
  return (
    <Flip right>
      <div style={myBarBackGround}>
        <div style={myBarForGround}></div>
      </div>
    </Flip>
  )
}
