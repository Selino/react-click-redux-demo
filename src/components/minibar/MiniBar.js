import React from "react"
import colors from "../../tokens/colors"

export default function MiniBar(props) {
  let backGroundColor = props.backGroundColor
    ? props.backGroundColor
    : colors.graysGray4
  let forGroundColor = props.forGroundColor
    ? props.forGroundColor
    : colors.primaryInactive
  const myBarBackGround = {
    backgroundColor: backGroundColor,
    width: "60px",
    height: "16px",
    borderRadius: "2px",
    overflow: "hidden",
    display: "inline-block",
    marginLeft: ".5rem",
    verticalAlign: "text-bottom",
  }
  const myBarForGround = {
    backgroundColor: forGroundColor,
    width: props.Perc + "%",
    height: "100%",
  }
  return (
    <div style={myBarBackGround}>
      <div style={myBarForGround}></div>
    </div>
  )
}
