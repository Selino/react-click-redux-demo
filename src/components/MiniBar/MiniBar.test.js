import React from "react"
import { render, screen, fireEvent } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import MiniBar from "./MiniBar"

console.warn = jest.fn()

describe("MiniBar", () => {
  test("should match snapshot at 50% width", () => {
    const { asFragment } = render(
      <MiniBar
        backGroundColor='rgb(153, 153, 153)'
        forGroundColor='rgb(204, 204, 204)'
        Perc='50'
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("should match snapshot at 33% width", () => {
    const { asFragment } = render(
      <MiniBar
        backGroundColor='rgb(153, 153, 153)'
        forGroundColor='rgb(204, 204, 204)'
        Perc='33'
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
