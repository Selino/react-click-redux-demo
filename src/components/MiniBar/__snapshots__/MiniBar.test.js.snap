// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniBar should match snapshot at 33% width 1`] = `
<DocumentFragment>
  <div
    style="background-color: rgb(153, 153, 153); width: 60px; height: 16px; border-radius: 2px; overflow: hidden; display: inline-block; margin-left: .5rem; vertical-align: text-bottom;"
  >
    <div
      style="background-color: rgb(204, 204, 204); width: 33%; height: 100%;"
    />
  </div>
</DocumentFragment>
`;

exports[`MiniBar should match snapshot at 50% width 1`] = `
<DocumentFragment>
  <div
    style="background-color: rgb(153, 153, 153); width: 60px; height: 16px; border-radius: 2px; overflow: hidden; display: inline-block; margin-left: .5rem; vertical-align: text-bottom;"
  >
    <div
      style="background-color: rgb(204, 204, 204); width: 50%; height: 100%;"
    />
  </div>
</DocumentFragment>
`;
